packages/components/src/tooltip/index.js

```diff
@@ -109,6 +151,11 @@ function Tooltip( props ) {
 	const [ isMouseDown, setIsMouseDown ] = useState( false );
 	const [ isOver, setIsOver ] = useState( false );
 	const delayedSetIsOver = useDebounce( setIsOver, delay );
+	const childRef = useRef( null );
+	const mergedChildRefs = useMergeRefs( [
```

Comment:
Maybe we could add some inline code to explain why we're storing both `childRef` and `mergedChildRefs` (which are being used later in code in the `getElementWithPopover` function)

Reply:
We need the first child's ref now because `cloneElement` would previously don't update the `ref`, correct? Also can you explain a bit why would we need the `childRef` in `getRegularElement`? Just trying to understand it fully 😄

Reply:
This one was the result of trial and error so I'm not 100% sure how it's working but here's my current understanding:

* We need a stable ref to the first child to be passed to the Popover so that it can calculate the correct position. In order to pass it reliably to the Popover without causing any errors, by using an additional `childRef` we have the one reference that is pointing to the current child element.
* Because we can't know whether or not the first child of the Tooltip is already using a `ref` for another reason, we need to attempt to merge it in with the `childRef` we're using for the `anchorRef`, this way we don't accidentally overwrite a `ref` used for other purposes.
* In `getRegularElement` and `getDisabledElement` we specify the mergedRef so that both refs are attached correctly to the cloned element (any existing ref is preserved, and our additional `childRef` for the Popover can be used for position calculation)

Let me know if you think any changes are needed here, or if we can make it clearer what's going on in the inline comment!

Reply:
Thanks for explaining! I'm afraid I can't suggest something right now, since I tried doing what I asked above, that is pass only the first child's ref if exists to `cloneElement` and only pass `childRef` var as the `anchorRef` without success.. It would be great though if we were 100% sure for the solution --cc @ellatrix @diegohaz if they have any insights.

Reply:
I'm having a hard time understanding what's going on here, but it's definitely good practice to create new refs for each purpose and not try to reuse refs created elsewhere. So the use of `useMergeRefs` seems fine here, if that's what you're asking.
Seeing `Children.toArray( children )[ 0 ]?.ref` makes me frown a little though. Can't you use the `child` variable that you have as well?

Reply:
> So the use of useMergeRefs seems fine here, if that's what you're asking.

Thanks for confirming, I think that gets at the main issue?

> Seeing Children.toArray( children )[ 0 ]?.ref makes me frown a little though. Can't you use the child variable that you have as well?

This is an interesting one — I had a go at consolidating with the exiting `child` variable, but it results in a few subtle issues:
* React's `Children.only()` function [throws an error](https://reactjs.org/docs/react-api.html#reactchildrenonly) rather than returning `undefined` so we can't use it before the conditional to check how many children there are. 
* Because we're using a hook here, we have the call to the hook after a conditional.
* If we switch `child` to use the `Children.toArray( children )[ 0 ]` approach, then an incorrect `key` gets copied when the element is duplicated, which causes other components' tests to fail (I found that the hard way in https://github.com/WordPress/gutenberg/pull/41268/commits/90f5541074b829d40000979bb18d3ffcba925043)

So, as far as I can tell the best trade-off we have so far is this slightly awkward way of accessing the child in order to retrieve the existing `ref` (if it exists). I've tidied this up a tiny bit by storing the value in `const existingChildRef` to see if that helps with clarity.

Let me know if either of you can think of a better way to handle this, I've very rarely had to use `React.Children` directly so I am a little unfamiliar with this part of the API. I'm happy to keep iterating now or in follow-ups.